pkg/kubectl/cluster.go

```diff
@@ -30,6 +31,8 @@ import (
 type ClusterGeneratorV1Beta1 struct {
 	// Name of the cluster context (required)
 	Name string
+	// Name of the federation this cluster is joining (required)
```

Comment:
Hmm. Do you mean that you can't use an old version of kubefed with a newer federation? Or vice versa? Can you give an example of what would fail that we need to continue supporting?

On a side note, do we have a grid for which versions of kubefed, the FCP, the host cluster Kubernetes version and the joined cluster versions we need to support, in new federation and upgrade cases? I know that we want to maintain backwards compatibility, but I'm not sure I really understand what sort of compatibility we're trying to maintain.

Adding the annotation does give you the ability to get all objects related to a particular federation, rather than having to get all the annotated objects and then the cluster objects separately.

That all said, I'm happy to remove the cluster annotations. I just want to make sure I understand the benefit.